react/src/components/SettingPage.tsx

```diff
@@ -0,0 +1,58 @@
+import Flex from './Flex';
+import SettingPageItem from './SettingPageItem';
+import { SettingItemProps } from './SettingPageItem';
+import { SearchOutlined } from '@ant-design/icons';
+import { List, Typography, Input } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+//interface for settings
+interface SettingPageProps {
+  settingOptions: [string, SettingItemProps[]][];
+}
+
+const SettingPage: React.FC<SettingPageProps> = ({
+  settingOptions,
+}: SettingPageProps) => {
+  const [searchValue, setSearchValue] = useState('');
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <Flex direction="column" style={{ paddingTop: 16 }}>
+        <Input
+          prefix={<SearchOutlined />}
+          //Todo: change the placeholder text
+          placeholder={t('table.SearchTableColumn')}
+          onChange={(e) => setSearchValue(e.target.value)}
+          value={searchValue}
+        />
+      </Flex>
+      <Flex direction="column" align="start">
+        <List
```

Comment:
If you want to support the grouping feature for items, how about using another React component such as [virtuoso](https://virtuoso.dev/grouped-numbers/) instead of Ant Design's List?